### PR TITLE
Add field `UnconfirmedTransfers` to RPC endpoint `ListAssets`

### PR DESCRIPTION
**Pull request by ffranr**
Added a new field `UnconfirmedTransfers` to the `ListAssets` RPC endpoint response. The field is a count of unconfirmed transfers. This field helps users interpret asset list responses, especially when asset coins are unconfirmed.

This change will hopeful help users understand this sort of situation: https://github.com/lightninglabs/taproot-assets/issues/964

### CODE REVIEW

File: rpcserver.go [hunk 1]
```diff
@@ -957,8 +957,25 @@ func (r *rpcServer) ListAssets(ctx context.Context,
 		return nil, err
 	}
 
+	// We will also report the number of unconfirmed transfers. This is
+	// useful for clients as unconfirmed asset coins are not included in the
+	// asset list.
+	transfersResp, err := r.ListTransfers(
```
Comment (guggero) [inline rpcserver.go:963]: By default, `ListTransfers` returns both confirmed and unconfirmed transfers.
So I think we should instead call into `r.cfg.AssetStore.QueryParcels(ctx, nil, true)` to get the unconfirmed transfers directly.

**Review by guggero — CHANGES_REQUESTED** [review]
Very nice, should help a lot with the user experience.
Just need to make sure we only count actually unconfirmed transfers, not all of them.

**Comment by dstadulis** [comment]
Existing issue that might be closed by this PR: https://github.com/lightninglabs/taproot-assets/issues/645

Would #969 support reporting `pendingchannels` in lnd?

**Comment by ffranr** [comment]
> Existing issue that might be closed by this PR: #645
> 
> Would #969 support reporting `pendingchannels` in lnd?

I think pending channels count towards the unconfirmed transfers counter that's being added in this PR.

**Review by guggero — APPROVED** [review]
Nice, LGTM :tada:

**Review by gijswijs — COMMENTED** [review]
Related to @guggero 's comment earlier (https://github.com/lightninglabs/taproot-assets/pull/969#discussion_r1652083113) I think that the Go Doc comment of `QueryParcels` is incorrect. 
https://github.com/lightninglabs/taproot-assets/blob/eeb8a5b1a7fa32fa0b40c6b04abfe7415853af50/tapdb/assets_store.go#L3020

It should probably read something like this:
```
// QueryParcels returns the set of confirmed and unconfirmed parcels or exclusively the unconfirmed parcels.
```

**Comment by ffranr** [comment]
> Related to @guggero 's comment earlier ([#969 (comment)](https://github.com/lightninglabs/taproot-assets/pull/969#discussion_r1652083113)) I think that the Go Doc comment of `QueryParcels` is incorrect.
> 
> https://github.com/lightninglabs/taproot-assets/blob/eeb8a5b1a7fa32fa0b40c6b04abfe7415853af50/tapdb/assets_store.go#L3020
> 
> It should probably read something like this:
> 
> ```
> // QueryParcels returns the set of confirmed and unconfirmed parcels or exclusively the unconfirmed parcels.
> ```

Yes. But I think I'm using `QueryParcels` correctly in this PR.

**Review by gijswijs — APPROVED** [review]
LGTM! 🎉

**Pull request merged**